src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/IncrementalBuildTest.cs

```diff
@@ -171,7 +171,7 @@ public Class2 ()
 						Assert.IsNotNull (libfoo, "libfoo.so should exist in the .apk");
 
 						Assert.AreEqual (so.TextContent ().Length, new FileInfo (Path.Combine (Root, libbuilder.ProjectDirectory, lib.IntermediateOutputPath,
-							"native_library_imports", "armeabi-v7a", "libfoo.so")).Length,
+							"nl", "armeabi-v7a", "libfoo.so")).Length,
```

Comment:
Smells like we should parameterize this test so that both `native_library_imports` (`$(UseShortFileNames)`=False)` and `nl` (`$(UseShortFileNames)`=True) are checked.